source/OrderedTypes/SupLattice.lagda

```diff
@@ -0,0 +1,286 @@
+
+Sup-Lattice : (𝓤 𝓦 𝓥 : Universe) → (𝓤 ⊔ 𝓦 ⊔ 𝓥) ⁺  ̇
+Sup-Lattice 𝓤 𝓦 𝓥 = Σ A ꞉ 𝓤  ̇ , sup-lattice-structure A
+ where
+  open sup-lattice-def 𝓤 𝓦 𝓥
+
+⟨_⟩ : Sup-Lattice 𝓤 𝓦 𝓥 → 𝓤  ̇
+⟨ A , rest ⟩ = A
+
+order-of : (L : Sup-Lattice 𝓤 𝓦 𝓥) → (⟨ L ⟩ → ⟨ L ⟩ → Ω 𝓦)
+order-of (A , (_≤_ , ⋁_) , rest) = _≤_
+
+syntax order-of L x y = x ≤⟨ L ⟩ y
+
+join-of : (L : Sup-Lattice 𝓤 𝓦 𝓥) → Fam 𝓥 ⟨ L ⟩ → ⟨ L ⟩
+join-of (A , (_≤_ , ⋁_) , rest) = ⋁_
+
+syntax join-of L U = ⋁⟨ L ⟩ U
+
+partial-orderedness-of : (L : Sup-Lattice 𝓤 𝓦 𝓥)
+                       → is-partial-order ⟨ L ⟩ (order-of L)
+partial-orderedness-of (A , (_≤_ , ⋁_) , order , is-lub-of) = order
+
+reflexivity-of : (L : Sup-Lattice 𝓤 𝓦 𝓥) → is-reflexive (order-of L) holds
+reflexivity-of L = pr₁ (pr₁ (partial-orderedness-of L))
+
+antisymmetry-of : (L : Sup-Lattice 𝓤 𝓦 𝓥) → is-antisymmetric (order-of L) 
+antisymmetry-of L = pr₂ (partial-orderedness-of L)
+
+transitivity-of : (L : Sup-Lattice 𝓤 𝓦 𝓥) → is-transitive (order-of L) holds
+transitivity-of L = pr₂ (pr₁ (partial-orderedness-of L))
+
+join-is-lub-of : (L : Sup-Lattice 𝓤 𝓦 𝓥)
+               → (U : Fam 𝓥 ⟨ L ⟩)
+               → ((order-of L) Joins.is-lub-of join-of L U) U holds
+join-is-lub-of (A , (_≤_ , ⋁_) , order , suprema) = suprema
+
+join-is-upper-bound-of : (L : Sup-Lattice 𝓤 𝓦 𝓥)
+                       → (U : Fam 𝓥 ⟨ L ⟩)
+                       → ((order-of L) Joins.is-an-upper-bound-of
+                                          join-of L U) U holds
+join-is-upper-bound-of L U = pr₁ (join-is-lub-of L U)
+
+join-is-least-upper-bound-of : (L : Sup-Lattice 𝓤 𝓦 𝓥)
+                             → (U : Fam 𝓥 ⟨ L ⟩)
+                             → ((u' , _) : Joins.upper-bound (order-of L) U)
+                             → (order-of L (join-of L U) u') holds
+join-is-least-upper-bound-of L U = pr₂ (join-is-lub-of L U)
+
+sethood-of : (L : Sup-Lattice 𝓤 𝓦 𝓥) → is-set ⟨ L ⟩
+sethood-of L =
+  type-with-prop-valued-refl-antisym-rel-is-set
+   (λ x → λ y → order-of L x y holds)
+   (λ x → λ y → holds-is-prop (order-of L x y))
+   (λ x → reflexivity-of L x)
+   (λ x → λ y → antisymmetry-of L)
+
+\end{code}
+
+We now define monotone endomaps on a sup-lattice. This notion is not too
+restrictive as our interest is with fixed points.
+
+\begin{code}
+
+module _ {𝓤 𝓦 𝓥 : Universe} (L : Sup-Lattice 𝓤 𝓦 𝓥) where
+
+ is-monotone : (f : ⟨ L ⟩ → ⟨ L ⟩) → 𝓤 ⊔ 𝓦  ̇
+ is-monotone f = (x y : ⟨ L ⟩) → (x ≤⟨ L ⟩ y) holds → (f x ≤⟨ L ⟩ f y) holds
+
+\end{code}
+
+We now show that when one subset contains another the join of their total
+spaces are ordered as expected. 
+
+\begin{code}
+
+module _ {𝓤 𝓦 𝓥 : Universe}
+         (L : Sup-Lattice 𝓤 𝓦 𝓥)
+         {A : 𝓥  ̇}
+         (m : A → ⟨ L ⟩)
+          where
+
+ open Joins (order-of L)
+
+ joins-preserve-containment : {P : 𝓟 {𝓥} A} {Q : 𝓟 {𝓥} A}
+                            → P ⊆ Q
+                            → ((⋁⟨ L ⟩ (𝕋 P , m ∘ 𝕋-to-carrier P))
```

Comment:
Since you're using Ayberk's `⋁⟨ L ⟩` which takes families only, I think you should make some boilerplate for turning subsets into families so that you don't have to write these pairs of the total space and its projection all the time.

Reply:
Good idea. The notation I went with is giving agda's type inference a bit of trouble. I may have you take a look.

Reply:
I tried to make L implicit when I returned to making it explicit it was fine... but this is a bit clunkier then I would like.